include/mqt-core/dd/Operations.hpp

```diff
@@ -22,83 +22,87 @@
 #include <vector>
 
 namespace dd {
-// single-target Operations
-template <class Config>
-qc::MatrixDD
-getStandardOperationDD(const qc::StandardOperation* op, Package<Config>& dd,
-                       const qc::Controls& controls, const qc::Qubit target,
-                       const bool inverse) {
-  GateMatrix gm;
 
-  const auto type = op->getType();
+inline std::tuple<qc::OpType, fp, fp, fp>
+getOperationParameters(const qc::StandardOperation* op, qc::Qubit* target0,
+                       qc::Qubit* target1, bool inverse) {
+  auto type = op->getType();
   const auto& parameter = op->getParameter();
 
   switch (type) {
-  case qc::I:
-    gm = I_MAT;
-    break;
-  case qc::H:
-    gm = H_MAT;
-    break;
-  case qc::X:
-    gm = X_MAT;
-    break;
-  case qc::Y:
-    gm = Y_MAT;
-    break;
-  case qc::Z:
-    gm = Z_MAT;
-    break;
-  case qc::S:
-    gm = inverse ? SDG_MAT : S_MAT;
-    break;
-  case qc::Sdg:
-    gm = inverse ? S_MAT : SDG_MAT;
-    break;
-  case qc::T:
-    gm = inverse ? TDG_MAT : T_MAT;
-    break;
-  case qc::Tdg:
-    gm = inverse ? T_MAT : TDG_MAT;
-    break;
-  case qc::V:
-    gm = inverse ? VDG_MAT : V_MAT;
-    break;
-  case qc::Vdg:
-    gm = inverse ? V_MAT : VDG_MAT;
-    break;
+    // clang-format off
+  case qc::iSWAP: case qc::iSWAPdg:
+  case qc::Peres: case qc::Peresdg:
+  case qc::S:     case qc::Sdg:
+  case qc::T:     case qc::Tdg:
+  case qc::V:     case qc::Vdg:
+  case qc::SX:    case qc::SXdg:
+  case qc::I: case qc::H: case qc::X: case qc::Y: case qc::Z:
```

Comment:
This is not yet completely correct and clang-tidy rightfully complains here.
I, H, X, Y, Z are all self inverse. So these need not be XORed.

I am also not a big fan of fall-through switch statements, so I'd rather have an explicit return in each of the blocks.